### PR TITLE
Move code of the icfy-analyze CircleCI task to a NPM script

### PR DESCRIPTION
**Pull request by jsnajdr**
In hope to get better error handling a recovery, I'm moving the commands for the `icfy-analyze` CircleCI task (the `npm run preanalyze-bundles`, running the `bin/analyze-icfy.js` script, moving files to artifacts folder) to a NPM script called `analyze-icfy`. That script chains the steps using `&&` operator, relying on exit codes.

That should be more reliable (or at least more comprehensible) than the `set +o errexit` bash option.

This also creates a `build-client-stats` NPM script that's used by both `analyze-bundles` and `analyze-icfy` scripts.

**For more context, see also:**
- `icfy-analyze` CI run where `preanalyze-bundles` failed with error and yet the task proceeded to run `bin/icfy-analyze.js` instead of terminating: https://circleci.com/gh/Automattic/wp-calypso/601462
- a [bug in `cross-env`](https://github.com/kentcdodds/cross-env/issues/228) I found and reported when investigating the above CI failure. The CI failure doesn't seem to involve a SIGINT signal anywhere, but yet it's the only way I see how a failed `webpack` build could lead to exit code 0.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,8 +59,8 @@
 		"npm": "^6.12.1"
 	},
 	"scripts": {
-		"preanalyze-bundles": "cross-env-shell NODE_ENV=production CALYPSO_ENV=production EMIT_STATS=true PROGRESS=true npm run -s build-client-evergreen",
-		"analyze-bundles": "webpack-bundle-analyzer client/stats.json public/evergreen -h 127.0.0.1 -p 9898 -s gzip",
+		"analyze-bundles": "npm run build-client-stats && webpack-bundle-analyzer client/stats.json public/evergreen -h 127.0.0.1 -p 9898 -s gzip",
+		"analyze-icfy": "npm run build-client-stats && node --max-old-space-size=4096 bin/icfy-analyze.js && mv client/stats.json client/chart.json \"$CIRCLE_ARTIFACTS/icfy\"",
```
Comment (sirreal) [inline package.json]: If there's an issue with `cross-env-shell` that results in a `0` exist status when we expect non-`0`, this type of chaining will fail to capture that, won't it?

File: package.json [hunk 2]
```diff
@@ -81,6 +81,7 @@
 		"build-client-both": "CONCURRENT_BUILDS=2 concurrently -c cyan -n \"fallback ,evergreen\" \"npm run build-client-fallback\" \"npm run build-client-evergreen\"",
 		"build-client-if-prod": "node -e \"process.env.CALYPSO_ENV === 'production' && process.exit(1)\" || npm run build-client-both",
 		"build-client-if-desktop": "node -e \"(process.env.CALYPSO_ENV === 'desktop' || process.env.CALYPSO_ENV === 'desktop-development') && process.exit(1)\" || npm run build-client-fallback",
+		"build-client-stats": "cross-env-shell NODE_ENV=production CALYPSO_ENV=production EMIT_STATS=true PROGRESS=true npm run build-client-evergreen",
```
Comment (sirreal) [inline package.json]: If we have issues with `cross-env-shell`, I think we can drop it here.

**Review by sirreal — APPROVED** [review]
I left a few questions, but this seems fine to try 👍 

ICFY job completed successfully.

File: package.json (hunk 1 above)
Comment (jsnajdr) [reply package.json]: Yes, that's true. I discovered the `cross-env` issue only after migrating the `icfy-analyze` task to a NPM script and debugging it.

In other words, I don't expect this PR to solve any of the problems, but the refactoring still seems like a good idea to me 🙂

File: package.json (hunk 2 above)
Comment (jsnajdr) [reply package.json]: We can drop `cross-env` completely, but it means dropping the remaining support for Windows that we have.

`cross-env` doesn't do anything useful on Unix systems: the shell already understands the `ENV_VAR=value command --to-run` syntax. It's useful for Windows: there, `cross-env` will parse the `ENV_VAR=value` prefixes, and sets up the environment and runs `command --to-run` in a way that Windows `cmd.exe` understands.

While removing `cross-env`, we could also get rid of `rimraf` and friends and replace them with `rm -rf`. That saves time where `npm rimraf` needs to download the package from the NPM registry.

Comment (sgomes) [reply package.json]: +1. I think it's reasonable to remove the ability for non-Unix developers to generate stats while we wait for the `cross-env-shell` fix.

**Review by sgomes — APPROVED** [review]
LGTM 👍 Seems like a good refactor to remove as much logic as possible from the CircleCI config, and everything still seems to be working both locally and on CI.

File: package.json (hunk 2 above)
Comment (sgomes) [reply package.json]: > We can drop `cross-env` completely, but it means dropping the remaining support for Windows that we have.

I'm not sure I'd remove it permanently, more as a temporary measure. I'm hoping that `cross-env-shell` will be fixed at some point, or perhaps we can find an earlier version of it without the bug.

Comment (jsnajdr) [reply package.json]: OK, I removed the `cross-env` usages in https://github.com/Automattic/wp-calypso/pull/39502/commits/02d3d34c9aff26e8c0bf713ab271e7cd9106226b

The `terser` issue and the swallowed exit codes can bite at any time, including a production build.

**Pull request merged**